cartographer_grpc/framework/rpc.h

```diff
@@ -39,17 +40,27 @@ class Rpc {
     const State state;
     Service* service;
     Rpc* rpc;
+    bool pending;
   };
 
   Rpc(int method_index, ::grpc::ServerCompletionQueue* server_completion_queue,
       const RpcHandlerInfo& rpc_handler_info, Service* service);
 
+  void OnRequest();
```

Comment:
The public interface of Rpc seems larger than necessary. Perhaps some of this should happen in Service?